### DISCUSSION TITLE
How to make custom/smart components

### DISCUSSION THREAD

**Discussion by dimaqq** [react-hook-form/react-hook-form#5921] [Q&A]
I'd like to create a custom input, that has:
* a primary selection
* an array of secondary things
* and the primary selection determines what secondary things are pre-filled and no longer editable

I've a demo using `redux-form`, and I'm hoping to migrate to this library instead:

![cats2](https://user-images.githubusercontent.com/662249/125381779-78d6a380-e3cf-11eb-9171-b3abe8620590.gif)

**Comment by bluebill1049** [comment]
I think this should be a controlled component, and it doesn't need to be related to a validation library like react hook form or redux form. 

react hook form should be only doing the validation, which can be used by https://react-hook-form.com/api/usecontroller or Controller.
